packages/web/src/components/track/PremiumConditionsPill.tsx

```diff
@@ -0,0 +1,47 @@
+import { PremiumConditions, formatUSDCWeiToUSDString } from '@audius/common'
+import { IconLock } from '@audius/stems'
+import cn from 'classnames'
+
+import LoadingSpinner from 'components/loading-spinner/LoadingSpinner'
+
+import styles from './PremiumConditionsPill.module.css'
+import { isPremiumContentUSDCPurchaseGated } from './helpers'
+
+const messages = {
+  unlocking: 'Unlocking',
+  locked: 'Locked'
+}
+
+export const PremiumConditionsPill = ({
+  premiumConditions,
+  unlocking
+}: {
+  premiumConditions: PremiumConditions
+  unlocking: boolean
+}) => {
+  const isPurchase = isPremiumContentUSDCPurchaseGated(premiumConditions)
+  const icon = unlocking ? (
+    <LoadingSpinner className={styles.spinner} />
+  ) : isPurchase ? null : (
+    <IconLock />
+  )
+
+  let message = null
+  if (unlocking) {
+    // Show only spinner when unlocking a purchase
+    message = isPurchase ? null : messages.unlocking
+  } else {
+    message = isPurchase
+      ? `$${formatUSDCWeiToUSDString(premiumConditions.usdc_purchase.price)}`
+      : messages.locked
+  }
+
+  const colorStyle = isPurchase ? styles.premiumContent : styles.gatedContent
+
+  return (
+    <div className={cn(styles.container, colorStyle)}>
```

Comment:
Is this not a button? I know it wasn't before but it seems like one now the way it looks...

Reply:
It's not clickable, per se. You can click pretty much anywhere on the track tile to trigger the "how to unlock" modal. So this is acting more like a label than a button.